### PR TITLE
Fix recordings

### PR DESCRIPTION
**Pull request by darkdarkdragon**
**What does this pull request do? Explain your changes. (required)**
Fixes absolute urls to the segments in the VOD playlists

**Specific updates (required)**

- when `recordObjectStoreUrl` set for object store correctly generate public urls for sessions specified in `previousSessions` field
 

**How did you test each of these updates (required)**
Added unit test

**Does this pull request close any open issues?**

**Checklist:**

- [ ] README and other documentation updated
- [ ] Node runs in OSX and devenv
- [ ] All tests in `./test.sh` pass

### CODE REVIEW

File: server/recordings_test.go [hunk 1]
```diff
@@ -16,6 +16,72 @@ import (
...
+			glog.Error("Error parsing URL: ", err)
+			w.WriteHeader(http.StatusForbidden)
+			return
+		}
+		w.Write([]byte(`{"manifestID":"playback01", "recordObjectStore": "memory://recstore5",
+		"recordObjectStoreUrl":"https://pub.test/",
+		"previousSessions":["sess1","sess2"]}`))
+	}))
+	defer whts.Close()
+	AuthWebhookURL = whts.URL
```
Comment (yondonfu) [inline server/recordings_test.go:41]: Let's make sure to reset this package level var when this test completes to ensure that other tests that depend on `AuthWebhookURL` start with a clean state.

i.e.

```golang
oldAuthWebhookURL = AuthWebhookURL
AuthWebhookURL = whts.URL
defer func() { AuthWebhookURL = oldAuthWebhookURL }()
```

File: server/recordings_test.go [hunk 2]
```diff
@@ -16,6 +16,72 @@ import (
...
+	jpl = core.NewJSONPlaylist()
+	jpl.InsertHLSSegment(&profile, 1, "sess3/testNode/P144p25fps16x9/3.ts", 2100)
+	bjpl, _ = json.Marshal(jpl)
+	msess3.SaveData("testNode/playlist_3.json", bjpl, nil)
+
+	resp = makeReq("GET", "/recordings/sess3/P144p25fps16x9.m3u8")
+	body, _ := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
+	assert.Equal(200, resp.StatusCode)
+	assert.Equal("#EXTM3U\n#EXT-X-VERSION:3\n#EXT-X-MEDIA-SEQUENCE:0\n#EXT-X-TARGETDURATION:2100\n#EXTINF:2100.000,\nhttps:/pub.test/sess1/testNode/P144p25fps16x9/1.ts\n#EXT-X-DISCONTINUITY\n#EXTINF:2100.000,\nhttps:/pub.test/sess2/testNode/P144p25fps16x9/2.ts\n#EXT-X-DISCONTINUITY\n#EXTINF:2100.000,\nhttps:/pub.test/sess3/testNode/P144p25fps16x9/3.ts\n#EXT-X-ENDLIST\n", string(body))
```
Comment (yondonfu) [inline server/recordings_test.go]: I see that this assertion passes, but shouldn't the URLs be be prefixed with `https://pub.test` as opposed to `https:/pub.test`?

**Review by yondonfu — CHANGES_REQUESTED** [review]
Is my understanding here correct?

If a user passes any of the manifestIDs associated with a previous session for a `/recordings` request then the response will always be the full playlist for all previous sessions regardless of the manifestID provided?

File: server/recordings_test.go (hunk 2 above)
Comment (darkdarkdragon) [reply server/recordings_test.go]: Nice catch. I'm again made mistake by using 'path.Join' on URLs 😞 . Fixed

File: server/recordings_test.go (hunk 1 above)
Comment (darkdarkdragon) [reply server/recordings_test.go:41]: Fixed

**Comment by darkdarkdragon** [comment]
@yondonfu 
> If a user passes any of the manifestIDs associated with a previous session for a /recordings request then the response will always be the full playlist for all previous sessions regardless of the manifestID provided?

No, only sessions previous for provided one (if user requests 'sess1' then VOD will contain 'sess1' and 'sess2').

File: common/util.go [hunk 3]
```diff
@@ -380,3 +380,10 @@ func RatPriceInfo(priceInfo *net.PriceInfo) (*big.Rat, error) {
 
 	return big.NewRat(priceInfo.PricePerUnit, pixelsPerUnit), nil
 }
+
+func JoinPaths(path1, path2 string) string {
```
Comment (yondonfu) [inline common/util.go]: Would `JoinURL(url, path string)` be a more descriptive name if this function is only intended to be used as an alternative to `path.Join()` when the first string param is a URL?

**Review by yondonfu — COMMENTED** [review]
> No, only sessions previous for provided one (if user requests 'sess1' then VOD will contain 'sess1' and 'sess2').

Ah ok I confused myself - the full playlist returned will combine the playlists for all previous sessions (specified in the webook) and the session requested by the user. So, in the test added with this PR the full playlist combines the playlists for sess1, sess2 and sess3 because sess1 and sess2 are specified as previous sessions and sess3 is requested by the user. And for the example in your comment, the full playlist combines the playlists for sess1 and sess2 because they are specified as previous sessions (with sess1 also requested by the user).

File: common/util.go (hunk 3 above)
Comment (darkdarkdragon) [reply common/util.go]: Fixed

**Comment by darkdarkdragon** [comment]
@yondonfu 
Yes. If user requests 'sess3' then webhook will contain ```"previousSessions":["sess1","sess2"]``` and VOD will contain all three sessions.
If user requests 'sess2' then `previousSessions` will contain only 'sess1' and VOD will contain only two sessions.

**Review by yondonfu — APPROVED** [review]
LGTM after squashing

**Pull request merged**